v2/tools/generator/internal/codegen/pipeline/create_arm_types.go

```diff
@@ -448,6 +496,66 @@ func (c *armTypeCreator) convertObjectPropertiesForARM(t *astmodel.ObjectType, i
 	return result, nil
 }
 
+func (c *armTypeCreator) visitARMTypeName(this *astmodel.TypeVisitor, it astmodel.TypeName, ctx interface{}) (astmodel.Type, error) {
+	// Allow json type to pass through.
+	if it == astmodel.JSONType {
+		return it, nil
+	}
+
+	// Look up the definition
+	def, ok := c.definitions[it]
```

Comment:
minor: I think there's a `c.definitions.GetDefinition` that gets you a standard error here, rather than you having to craft it yourself.

Reply:
Thanks!